l10n_br_account/models/l10n_br_account.py

```diff
@@ -461,3 +461,57 @@ class L10n_brTaxDefinition(object):
     company_id = fields.Many2one('res.company', string='Company',
                                  related='tax_id.company_id',
                                  store=True, readonly=True)
+
+
+class AccountInvoiceTax(models.Model):
+
+    _inherit = "account.invoice.tax"
+    
+    deduction_account_id = fields.Many2one('account.account','Tax Account for Deduction')
+    
+     
+    @api.v8
+    def compute(self, invoice):
+        result = super(AccountInvoiceTax,self).compute(invoice)
+        for line in invoice.invoice_line:
+            taxes = line.invoice_line_tax_id.compute_all(
+                (line.price_unit * (1 - (line.discount or 0.0) / 100.0)),
+                line.quantity, line.product_id, invoice.partner_id)['taxes']
+            for tax in taxes:
+                key = (tax['tax_code_id'], tax['base_code_id'], tax['account_collected_id'] or line.account_id.id)
+                if key in result.keys():
+                    if invoice.type in ('out_invoice','in_invoice'):
+                        result[key]['deduction_account_id'] = tax.get('account_deduced_id',False)
+                    else:
+                        result[key]['deduction_account_id'] = tax.get('account_paid_deduced_id',False)
+        return result
+     
+    @api.v7
+    def compute(self, cr, uid, invoice_id, context=None):
+        recs = self.browse(cr, uid, [], context)
+        invoice = recs.env['account.invoice'].browse(invoice_id)
+        return recs.compute(invoice)
+    
+    @api.model
+    #create counterpart credit tax
+    def move_line_get(self, invoice_id):
+        res = super(AccountInvoiceTax, self).move_line_get(invoice_id)
+        inv = self.env['account.invoice'].browse(invoice_id)
+        self._cr.execute(
+            'SELECT * FROM account_invoice_tax WHERE invoice_id = %s',
+            (invoice_id,)
+        )
+        for row in self._cr.dictfetchall():
```

Comment:
Why load inv and after do a 'select * ' in the same table, I think it's better in this case not bypass the ORM